### PR TITLE
Add "text" (or "label") button style/variant

### PR DESCRIPTION
**Pull request by ScopeyNZ**
Required for the new design for the delete button in add product

### CODE REVIEW

**Comment by robbieaverill** [comment]
Linting failure

File: src/components/ChecButton.vue [hunk 1]
```diff
@@ -39,18 +39,21 @@ export default {
...
     /**
-     * The style variant of the button. Includes size, border radius, etc. One of 'large', 'regular', 'small',
-     * 'round', 'tag'
+     * The style variant of the button. Includes size, border radius, etc. One of 'regular', 'small', 'round', 'tag'
      */
     variant: {
       type: String,
       validate(variant) {
-        return ['large', 'regular', 'small', 'round', 'tag'].includes(variant);
+        return ['regular', 'small', 'round', 'text', 'tag'].includes(variant);
```
Comment (robbieaverill) [inline src/components/ChecButton.vue:56]: Did you mean to remove large?

Is it used anywhere in the dashboard or auth service?

Comment (ScopeyNZ) [reply src/components/ChecButton.vue:56]: Oh yeah, I forgot to add that to the description. It's not used, and has been removed from Figma.

**Comment by ScopeyNZ** [comment]
Any further feedback?

**Comment by robbieaverill** [comment]
Any reason you didn't make this a variant?

The "brand" colour doesn't look right with the text colour:
image

**Comment by kvisca** [comment]
Seems weird to me to be a prop called text. If I saw that in the mark up I would think it's the actual text of the button. Maybe rename it to `textVariant` or `textOnly`.  

Otherwise, LGTM minus the brand variant not having a text-only version.

**Comment by ScopeyNZ** [comment]
I didn't make it a variant as I figured it has to match the size of other buttons with different variants. The "variant" as-is should probably be called "size".

Fixed the brand colour issue.

I didn't really like the naming either - but I really struggled to come up with something better. I considered `text-variant` but it felt weird with the `variant` prop too.

I could do a prop called `style` with the options being `text` and `button` but that also felt like it conflicts with `variant`.

**Comment by ScopeyNZ** [comment]
I've updated it to `textOnly`.

**Comment by robbieaverill** [comment]
You're essentially replicating the ["link" button style](https://getbootstrap.com/docs/4.4/components/buttons/) from Bootstrap. Their naming works because it looks exactly like a link when it renders. I agree with Kyle that text (and textOnly) are not good names for this prop.

What about "outline" or something?

**Comment by kvisca** [comment]
> I didn't make it a variant as I figured it has to match the size of other buttons with different variants. The "variant" as-is should probably be called "size".
> 
> Fixed the brand colour issue.
> 
> I didn't really like the naming either - but I really struggled to come up with something better. I considered `text-variant` but it felt weird with the `variant` prop too.
> 
> I could do a prop called `style` with the options being `text` and `button` but that also felt like it conflicts with `variant`.

I think `variant` should be renamed `style` and the `textOnly` to `variant` with a type of text or button like you suggested. That makes a lot more sense.

**Comment by robbieaverill** [comment]
I agree with Kyle, but not "style" because that's a native HTML attribute name

**Comment by ScopeyNZ** [comment]
I agree that variant should be renamed, probably to `size` or something, but that's a lot more effort 😢 . I'd have to PR all the repos that use buttons to change everything. Do you think I can raise an issue to do that outside of this PR?

Happy to rename the existing prop to whatever makes the most sense until then though.

**Comment by kvisca** [comment]
> I agree that variant should be renamed, probably to `size` or something, but that's a lot more effort 😢 . I'd have to PR all the repos that use buttons to change everything. Do you think I can raise an issue to do that outside of this PR?
> 
> Happy to rename the existing prop to whatever makes the most sense until then though.

Yea I think as long as we have an issue to rename it we're good. A couple of components have this odd variant naming issue, so we could attack them all at once then.

**Pull request merged**